frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,384 @@
+<!-- BranchButton.svelte -->
+<script lang="ts">
+	import { branchName, documentTreeStore, currentFile, editorText } from '$lib/main';
+	import { derived, get } from 'svelte/store';
+	import { onMount } from 'svelte';
+	import { apiAddress } from '$lib/net';
+	import { ToastType, addToast } from '$lib/toast';
+	import { cache } from '$lib/cache';
+
+	let showMenu = false;
+	let protectedBranches: string[] = [];
+	let nonProtectedBranches: string[] = [];
+	let newBranchName: string = '';
+	let showInput = false;
+
+	const currentBranch = derived(branchName, ($branchName) => $branchName);
+
+	/**
+	 * Fetches existing branches from the GitHub API.
+	 *
+	 * This asynchronous function sends a GET request to retrieve the list of branches
+	 * from the specified GitHub repository. If the request is successful, it returns
+	 * a promise that resolves to an array containing the branch data. If the request
+	 * fails (e.g., due to network issues, authentication problems, or server errors),
+	 * it returns a promise that rejects with an error object containing details about the failure.
+	 *
+	 * @returns {Promise<Array>} A promise that resolves to an array of branches on success,
+	 * or rejects with an error object containing details about the failure.
+	 *
+	 * @throws Will throw an error if the response from the API is not successful. The error object
+	 * will include the status code and any additional error message from the API.
+	 *
+	 * @example
+	 * async function main() {
+	 *   try {
+	 *     const branches = await fetchExistingBranches();
+	 *     console.log('Existing branches:', branches);
+	 *   } catch (err) {
+	 *     console.error('Failed to fetch branches:', err);
+	 *   }
+	 * }
+	 */
+	async function fetchExistingBranches() {
```

Comment:
```suggestion
	async function fetchExistingBranches(): Promise<Branch[]> {
```

Note, you will need to define the type for a branch somewhere, ala <https://www.typescriptlang.org/docs/handbook/interfaces.html>
```ts
interface Branch {
  <insert your declaration here>
}

```